infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/WaitSnapshotApplyState.java

```diff
@@ -102,6 +109,14 @@ public LogReplicationState processEvent(LogReplicationEvent event) throws Illega
                  Snapshot Sync with ID = 1 could be completed in between (1 and 2) but show up in the queue
                  as 4, attempting to process a completion event for the incorrect snapshot sync.
                  */
+
+                if (tableManagerPlugin.isUpgraded()) {
```

Comment:
shouldnt we update this flag only after the check on L120?  Also transition to LOG_ENTRY_SYNC state?

Reply:
Fixed, thanks!